File: assiist_front_end/lib/core/models/reservation.dart
class Reservation {
  final String id;
  final String contactId;
  final String contactName;
  final String? contactEmail;
  final String? contactPhone;
  final String userId;
  final String accountId;
  final DateTime createdOn;
  final DateTime updatedOn;

  Reservation({
    required this.id,
    required this.contactId,
    required this.contactName,
    this.contactEmail,
    this.contactPhone,
    required this.userId,
    required this.accountId,
    required this.createdOn,
    required this.updatedOn,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      id: json['id'] as String,
      contactId: json['contact_id'] as String,
      contactName: json['contact_name'] as String,
      contactEmail: json['contact_email'] as String?,
      contactPhone: json['contact_phone'] as String?,
      userId: json['user_id'] as String,
      accountId: json['account_id'] as String,
      createdOn: DateTime.parse(json['created_on'] as String),
      updatedOn: DateTime.parse(json['updated_on'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'contact_id': contactId,
      'contact_name': contactName,
      'contact_email': contactEmail,
      'contact_phone': contactPhone,
      'user_id': userId,
      'account_id': accountId,
      'created_on': createdOn.toIso8601String(),
      'updated_on': updatedOn.toIso8601String(),
    };
  }
}

File: assiist_back_end/api/schemas/reservation.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field
from enum import Enum

class ReservationStatus(str, Enum):
    """Possible statuses for a reservation."""
    ON_LIST = "on_list"
    OFF_LIST = "off_list"

class ReservationBaseSchema(BaseModel):
    """Base schema for reservation data."""
    contact_id: str = Field(..., description="ID of the contact this reservation is for")
    account_id: str = Field(..., description="ID of the account this reservation belongs to")
    contact_name: str = Field(..., description="Name of the contact")
    contact_email: Optional[str] = Field(None, description="Email of the contact")
    contact_phone: Optional[str] = Field(None, description="Phone number of the contact")

class ReservationCreateSchema(ReservationBaseSchema):
    """Schema for creating a new reservation."""
    pass

class ReservationUpdateSchema(BaseModel):
    """Schema for updating an existing reservation."""
    contact_name: Optional[str] = Field(None, description="Updated name of the contact")
    contact_email: Optional[str] = Field(None, description="Updated email of the contact")
    contact_phone: Optional[str] = Field(None, description="Updated phone number of the contact")

class ReservationResponseSchema(ReservationBaseSchema):
    """Schema for reservation responses."""
    id: str = Field(..., description="Unique identifier for the reservation")
    user_id: str = Field(..., description="ID of the user who created the reservation")
    created_on: datetime = Field(..., description="When the reservation was created")
    updated_on: datetime = Field(..., description="When the reservation was last updated")
    created_by: str = Field(..., description="ID of the user who created the reservation")
    updated_by: str = Field(..., description="ID of the user who last updated the reservation")

    class Config:
        from_attributes = True 